mkosi/run.py

```diff
@@ -244,6 +244,12 @@ def preexec() -> None:
             # expect it to pick.
             assert nfd == SD_LISTEN_FDS_START + i
 
+    if (
+        (not sandbox or subprocess.run(sandbox + ["true"]).returncode == 0) and
```

Comment:
Maybe it's a bit late, but this line confuses me. I assume the `true` check is to see that something can run successfully in the sandbox if one exists? But if that's not the case (the rc isn't 0), then the `die` will be skipped and the rest of the function will probably fail for other reasons. Wouldn't it be better to fail earlier? Also in that case the whole line could go, because it wouldn't matter whether sandbox were empty or not.